src/Context.h

```diff
@@ -115,6 +120,25 @@ class Context
   long                                time_sort_us        {0};
   long                                time_render_us      {0};
   long                                time_hooks_us       {0};
+
+  // the current task for DOM references, or NULL if there is no task
+  const Task *                        currentTask         {NULL};
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// CurrentTask resets Context::currentTask to NULL on destruction; this ensures
+// that this context value is not a dangling pointer.
```

Comment:
As far as I can tell, technically currently we the implementation does "stack" the contextTask references (similar to nesting Python's contextmanagers), so perhaps we should adjust the comment here to reflect that?

In other words:
```suggestion
// CurrentTask resets Context::currentTask to previous context task on destruction; this ensures
// that this context value is restored when exiting the scope where the context was applied.
```